kernel/src/mm/phys/frame.rs

```diff
@@ -176,4 +208,49 @@ impl FrameAllocator {
 
         Ok(())
     }
+
+    ///
+    /// # Description
+    ///
+    /// Allocates all frames as possible in range.
+    ///
+    /// # Parameters
+    ///
+    /// - `start`: Start page frame address.
+    /// - `end`: End page frame address.
+    ///
+    /// # Returns
+    ///
+    /// Upon success, `Ok(())` is returned. Upon failure, an error is returned instead.
+    ///
+    pub fn try_alloc_range(
+        &mut self,
+        region: &TruncatedMemoryRegion<PhysicalAddress>,
+    ) -> Result<(), Error> {
+        let start_frame_number: usize = region.start().into_frame_number().into_raw_value();
+        let end_frame_number: usize = start_frame_number + region.size() / mem::FRAME_SIZE - 1;
+
+        // Try book all frames in the range.
+        for index in start_frame_number..=end_frame_number {
+            match self.bitmap.set(index) {
+                Ok(()) => continue,
+                Err(e) if e.code == ErrorCode::ResourceBusy => continue,
+                Err(e) => {
+                        warn!("fail to try book frame {}", index);
+                        return Err(e);
+                }
+            }
+        }
```

Comment:
If we fail in the middle and return `Err()` we would leave the system in an inconsistent state (some bits were set).

To fix this we should either:
1) Rollback what was allocated.
2) Before allocating, checking if resources are available.